### DISCUSSION TITLE
ColorMode doesn't work in Remix

### DISCUSSION THREAD

**Discussion by kupuma-ru21** [chakra-ui/chakra-ui#9461] [General]
### Description

I use Remix with Chakra so I did "[Using Chakra in Remix](https://www.chakra-ui.com/docs/get-started/frameworks/remix)"

After that, I put "[ColorModeButton](https://www.chakra-ui.com/docs/styling/dark-mode#adding-the-dark-mode-toggle)" into a page and I clicked the button
But nohting happens
My exptections are when I click the button, color mode changes per click

As an FYI
If I don't change "entry.server.tsx", ColorMode works as I expected but I get a warning

|           not working           |           warning            |
| :------------------------: | :------------------------: |
|<video src="https://user-images.githubusercontent.com/https://github.com/user-attachments/assets/5d3255f7-4939-4a4b-ab84-27f28332f7f4">|<video src="https://user-images.githubusercontent.com/https://github.com/user-attachments/assets/1668d0d1-b9b9-4ec1-a38f-5842aca460bd">|

### Link to Reproduction

https://github.com/kupuma-ru21/chakra-with-remix

### Steps to reproduce

2. Click on '...'
3. Scroll down to '...'

### Chakra UI Version

3.0.2

### Browser

Google chrome

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by thesohailjafri** [comment]
I reproduced the code, and the style is working fine in Remix, but it doesn’t work as expected for JavaScript events. For example, I added a button with a `colorPalette` set to `blue`, and the styling worked. However, when I attached an onClick event to console something or trigger an alert, it didn’t work.

![image](https://github.com/user-attachments/assets/e18920e9-0b06-44de-bb48-8f3bf6be2a8d)

```jsx
import { Stack, Text } from '@chakra-ui/react'
import type { MetaFunction } from '@remix-run/node'
import { Button } from '~/components/ui/button'
import { ColorModeButton } from '~/components/ui/color-mode'

export const meta: MetaFunction = () => {
  return [
    { title: 'New Remix App' },
    { name: 'description', content: 'Welcome to Remix!' },
  ]
}

export default function Index() {
  return (
import { Container, Stack, Text } from '@chakra-ui/react'
import type { MetaFunction } from '@remix-run/node'
import { Button } from '~/components/ui/button'
import { ColorModeButton } from '~/components/ui/color-mode'

export const meta: MetaFunction = () => {
  return [
    { title: 'New Remix App' },
    { name: 'description', content: 'Welcome to Remix!' },
  ]
}

export default function Index() {
  return (
    <Container py={10}>
      <Stack>
        <Text>
          hello world, this below is a button that toggles the color mode
        </Text>
        <ColorModeButton />
        <Text>
          hello world, this below is a button that logs a message to the console
        </Text>
        <Button
          colorPalette={'blue'}
          width={'fit-content'}
          onClick={() => {
            console.log('You clicked me!')
            alert('You clicked me!')
          }}
        >
          Click me
        </Button>
      </Stack>
    </Container>
  )
}
```

**Comment by vincentes** [comment]
Any update on this? I am having the same issue.

**Comment by segunadebayo** [comment]
Closing as stale. Please start a new discussion if this is still relevant on the latest version.
